dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,799 @@
+                output_file = os.path.join(
+                    self.write_dir,
+                    "%s_csd_%d_%s.h5" % (src.lower(), csd, self.output_suffix),
+                )
+                self.logger.info("Writing {}...".format(output_file))
+
+                self.update_data_index(
+                    data.time[0], data.time[-1], filename=output_file
+                )
+
+                with h5py.File(output_file, "w") as handler:
+
+                    index_map = handler.create_group("index_map")
+                    index_map.create_dataset("freq", data=data.index_map["freq"][:])
+                    index_map.create_dataset("pol", data=np.string_(pols))
+                    index_map.create_dataset("ha", data=ha)
+
+                    dset = handler.create_dataset("ra", data=ra)
+                    dset.attrs["axis"] = np.array(["ha"], dtype="S")
+
+                    dset = handler.create_dataset("time", data=data.time)
+                    dset.attrs["axis"] = np.array(["ha"], dtype="S")
+
+                    dset = handler.create_dataset("vis", data=vis)
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    dset = handler.create_dataset("peak_vis", data=peak_flux)
+                    dset.attrs["axis"] = np.array(["freq", "pol"], dtype="S")
+
+                    dset = handler.create_dataset(
+                        "weight", data=tools.invert_no_zero(var)
+                    )
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    dset = handler.create_dataset("count", data=counter.astype(np.int))
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    index_map.create_dataset("param", data=fitter.parameter_names)
+
+                    dset = handler.create_dataset("residual_noise", data=resid_rms)
+                    dset.attrs["axis"] = np.array(["freq", "pol"], dtype="S")
+
+                    dset = handler.create_dataset("parameter", data=fitter.param)
+                    dset.attrs["axis"] = np.array(["freq", "pol", "param"], dtype="S")
+
+                    dset = handler.create_dataset("chisq", data=fitter.chisq)
+                    dset.attrs["axis"] = np.array(["freq", "pol"], dtype="S")
+
+                    dset = handler.create_dataset("ndof", data=fitter.ndof)
+                    dset.attrs["axis"] = np.array(["freq", "pol"], dtype="S")
+
+                    dset = handler.create_dataset(
+                        "parameter_cov", data=fitter.param_cov
+                    )
+                    dset.attrs["axis"] = np.array(
+                        ["freq", "pol", "param", "param"], dtype="S"
+                    )
+
+                    handler.attrs["source"] = src
+                    handler.attrs["csd"] = csd
+                    handler.attrs["chisq"] = fitter.chisq
+                    handler.attrs["ndof"] = fitter.ndof
```

Comment:
You can remove `chisq` and `ndof` from the attributes now that there are datasets.